### PR TITLE
Refactoring to TypeScript and Project Configuration Updates

### LINKED ISSUE #14: Migrate the initiated server from js to ts.
You can refer here for the migration - 
https://medium.com/@mhuckstepp/step-by-step-guide-to-convert-an-existing-express-node-js-backend-to-typescript-931e435ea95d
https://blog.phillipninan.com/how-to-convert-expressjs-to-typescripthttps://blog.phillipninan.com/how-to-convert-expressjs-to-typescript

### PR DESCRIPTION
**Pull request by ad-astra-per-ardua**
image

**Summary of Changes:**

1. index.js and routes.js have been transformed to index.ts and routes.ts, respectively.
2. TypeScript-related settings have been added to tsconfig.json.
3. Several packages have been added to package.json.

**Detailed Changes:**

TypeScript Refactoring:

- The files index.js and routes.js have been changed to .ts extensions.
- Explicit types have been specified for the Express application and router variables.

TypeScript Configuration:

- The tsconfig.json file has been added to configure the TypeScript project.
- Key configurations include the compilation target version, libraries used, and the module system.

Dependency Updates:

- Several frontend-related version packages and the typescript package have been added.
- TypeScript-related scripts have been added to the execution scripts.

I kindly ask for a thorough review. Please share any thoughts or suggestions you might have. :>

### CODE REVIEW

**Comment by ad-astra-per-ardua** [comment]
#14

**Comment by AmreetKumarkhuntia** [comment]
Will check it out

File: server/src/index.ts [hunk 1]
```diff
@@ -21,13 +21,14 @@ app.use(express.json());
...
 
 app.get("/", (req, res) => {
-  res.json("Server is running!");
+    res.send("Server is running!<br>Looks everything's fine.");
 });
 
 // Start the Express server
 app.listen(port, () => {
   console.log(`Server is running on port ${port}`);
+  console.log(`Refactoring Completed.`);
```
Comment (AmreetKumarkhuntia) [inline server/src/index.ts]: Remove this console

File: server/package.json [hunk 2]
```diff
@@ -1,23 +1,62 @@
...
+    "@types/react": "^18.2.28",
+    "@types/react-dom": "^18.2.13",
     "cors": "^2.8.5",
     "dotenv": "^16.3.1",
-    "express": "^4.18.2",
+    "framer-motion": "^10.16.4",
     "helmet": "^7.0.0",
     "morgan": "^1.10.0",
-    "nodemon": "^3.0.1"
+    "react": "^18.2.0",
```
Comment (AmreetKumarkhuntia) [inline server/package.json]: Remove react dependencies.

File: server/package.json [hunk 3]
```diff
@@ -1,23 +1,62 @@
 {
```
Comment (AmreetKumarkhuntia) [inline server/package.json]: looks like this is copied from client

**Review by AmreetKumarkhuntia — COMMENTED** [review]
Do not replace the package.json from client package

**Review by AmreetKumarkhuntia — CHANGES_REQUESTED** [review]
Work on the comments and do not replace the package.json file.

**Comment by ad-astra-per-ardua** [comment]
> Work on the comments and do not replace the package.json file.

Is it okay to change the start script from "start": "node src/index.ts" to "start": "ts-node src/index.ts" in our development setup?

**Comment by AmreetKumarkhuntia** [comment]
> > Work on the comments and do not replace the package.json file.
> 
> Is it okay to change the start script from "start": "node src/index.ts" to "start": "ts-node src/index.ts" in our development setup?

I don't think it's necessary but ya can keep it. Also, add the required dev dependencies for that to work.

**Comment by ad-astra-per-ardua** [comment]
image
image

- Adjusted the tsconfig.json to better suit the server environment and the use of TypeScript.
- Updated the package.json to reflect the necessary dependencies and scripts for TypeScript. Removed any unrelated dependencies, ensuring it's distinct from the client's package.json.

**Review by AmreetKumarkhuntia — APPROVED** [review]

**Comment by ad-astra-per-ardua** [comment]
It was honored :)

**Pull request merged**